### PR TITLE
Added ULID rule

### PR DESCRIPTION
**Pull request by AdamWills**

### CODE REVIEW

**Comment by aidan-casey** [comment]
Thanks for another PR, @AdamWills! Can you speak to whether or not ULID's can be lowercase according to the spec?

**Comment by didi955** [comment]
ULID are case insensitive according to https://github.com/ulid/spec

**Comment by aidan-casey** [comment]
@AdamWills - Do you mind updating accordingly?

File: src/Validation/Rules/Ulid.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+use Attribute;
+use Tempest\Validation\Rule;
+
+#[Attribute]
+final readonly class Ulid implements Rule
+{
+    public function isValid(mixed $value): bool
+    {
+        return preg_match('/^[0-9A-HJKMNP-TV-Z]{26}$/', $value) === 1;
```
Comment (didi955) [inline src/Validation/Rules/Ulid.php:15]: return preg_match('/^[0-9A-HJKMNP-TV-Z]{26}$/', strtoupper($value)) === 1;

File: tests/Validation/Rules/UlidTest.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+use Tempest\Validation\Rules\Ulid;
+
+class UlidTest extends TestCase
+{
+    public function test_ulid()
+    {
+        $rule = new Ulid();
+
+        $this->assertSame('Value should be a valid ULID', $rule->message());
+
```
Comment (didi955) [inline tests/Validation/Rules/UlidTest.php:17]: add a test for lowercase ULID

**Review by didi955 — CHANGES_REQUESTED** [review]

File: src/Validation/Rules/Ulid.php (hunk 1 above)
Comment (aidan-casey) [reply src/Validation/Rules/Ulid.php:15]: @didi955 - Why opt for `strtoupper` over including those characters in the expression?

**Comment by AdamWills** [comment]
@aidan-casey - updated with case insensitive matching!

**Comment by aidan-casey** [comment]
Thank you, @AdamWills!

**Pull request merged**